modules/solid_mechanics/src/actions/CommonSolidMechanicsAction.C

```diff
@@ -29,8 +31,98 @@ CommonSolidMechanicsAction::CommonSolidMechanicsAction(const InputParameters & p
 void
 CommonSolidMechanicsAction::act()
 {
+
   // check if sub-blocks block are found which will use the common parameters
-  auto action = _awh.getActions<QuasiStaticSolidMechanicsPhysicsBase>();
-  if (action.size() == 0)
+  auto actions = _awh.getActions<QuasiStaticSolidMechanicsPhysicsBase>();
+  if (actions.size() == 0)
     mooseWarning("Common parameters are supplied, but not used in ", parameters().blockLocation());
+
+  // Add disp-variables
+  if (_current_task == "add_variable")
+  {
+
+    // helper variables to collect what the user wants
+    bool do_add_variables = isParamSetByUser("add_variables") && getParam<bool>("add_variables");
+    std::vector<VariableName> displacement_variables;
+    if (isParamSetByUser("displacements"))
+    {
+      const auto v = getParam<std::vector<VariableName>>("displacements");
+      if (v.size() > 0)
+        displacement_variables.insert(displacement_variables.end(), v.begin(), v.end());
+    }
+    bool add_variables_block_restricted = true;
+    std::set<SubdomainName> add_variables_blocks;
+    bool scaling_set = isParamSetByUser("scaling");
+    Real scaling_value = (scaling_set) ? getParam<Real>("scaling") : 1.0;
+
+    // check all sub-actions
+    for (const auto & action : actions)
+    {
+      if (action->isParamSetByUser("add_variables") && action->getParam<bool>("add_variables"))
+      {
+        // this sub-action wants the disp-variables added.
+        do_add_variables = true;
+        const auto v = action->getParam<std::vector<VariableName>>("displacements");
+        if (v.size() > 0)
+        {
+          if (displacement_variables.size() == 0)
+            displacement_variables.insert(displacement_variables.end(), v.begin(), v.end());
+          else if (displacement_variables != v)
+            paramError("displacements",
+                       "The vector of displacement variables of the actions differ.");
+        }
```

Comment:
This error check might be too restrictive. Subblocks may declare displacements with different names, I don't see that necessarily as an error. So for example
```python
[Physics]
  [SolidMechanics]
    [QuasiStatics]
      [block1]
        displacements = 'foo_x foo_y foo_z'
        block = '1'
      []
      [block2]
        displacements = 'bar_x bar_y bar_z'
        block = '2'
      []
    []
  []
[]
```
I feel like this is fine...

Reply:
Oh nevermind me! I think we require a unique set of displacement variables for purpose of displaced mesh. This error check is perfectly fine then.